cupyx/scipy/ndimage/fourier.py

```diff
@@ -0,0 +1,194 @@
+import numpy as np
+
+import cupy
+from cupyx.scipy.ndimage import filters
+
+
+def _get_output_fourier(output, input):
+    if output is None:
+        if input.dtype.type in [cupy.complex64, cupy.complex128,
+                                cupy.float32]:
+            output = cupy.zeros(input.shape, dtype=input.dtype)
+        else:
+            output = cupy.zeros(input.shape, dtype=cupy.float64)
+    elif type(output) is type:
+        if output not in [cupy.complex64, cupy.complex128,
+                          cupy.float32, cupy.float64]:
+            raise RuntimeError("output type not supported")
+        output = cupy.zeros(input.shape, dtype=output)
+    elif output.shape != input.shape:
+        raise RuntimeError("output shape not correct")
+    return output
+
+
+def _get_output_fourier_complex(output, input):
+    if output is None:
+        if input.dtype.type in [cupy.complex64, cupy.complex128]:
+            output = cupy.zeros(input.shape, dtype=input.dtype)
+        else:
+            output = cupy.zeros(input.shape, dtype=cupy.complex128)
+    elif type(output) is type:
+        if output not in [cupy.complex64, cupy.complex128]:
+            raise RuntimeError("output type not supported")
+        output = cupy.zeros(input.shape, dtype=output)
+    elif output.shape != input.shape:
+        raise RuntimeError("output shape not correct")
+    return output
```

Comment:
I think you can combine the two functions?
```python
def _get_output_fourier(output, input, complex_only=False):
      types = [cupy.complex64, cupy.complex128]
      if not complex_only:
          types += [cupy.float32, cupy.float64]
     
      if output is None:
          if input.dtype in types:
              output = cupy.zeros(input.shape, dtype=input.dtype)
          else:
              output = cupy.zeros(input.shape, dtype=types[-1])
      elif type(output) is type:
          if output not in types:
              raise RuntimeError("output type not supported")
          output = cupy.zeros(input.shape, dtype=output)
      elif output.shape != input.shape:
          raise RuntimeError("output shape not correct")
      return output
```

Reply:
Yeah, I can refactor to combine some things.

Reply:
Thanks. I have now made the suggested change